### PR TITLE
refactor uses of UPath

### PR DESCRIPTION
**Pull request by normanrz**
Now that https://github.com/fsspec/universal_pathlib/pull/56, https://github.com/fsspec/universal_pathlib/pull/53 and https://github.com/fsspec/universal_pathlib/pull/52 are merged, we can simplify our code a bit.

### CODE REVIEW

File: webknossos/webknossos/dataset/_array.py [hunk 1]
```diff
@@ -141,7 +141,9 @@ def __init__(self, path: Path):
 
     @classmethod
     def open(cls, path: Path) -> "WKWArray":
-        if (path / "header.wkw").is_file():
+        header_path = path / "header.wkw"
+
+        if header_path.exists() and header_path.is_file():
```
Comment (normanrz) [inline webknossos/webknossos/dataset/_array.py:146]: This is a fix for a bug that @leowe found.

File: webknossos/webknossos/utils.py [hunk 2]
```diff
@@ -213,34 +212,21 @@ def warn_deprecated(deprecated_item: str, alternative_item: str) -> None:
     )
 
 
-def make_upath(maybe_path: Union[str, PathLike, Path]) -> UPath:
-    return maybe_path if isinstance(maybe_path, UPath) else UPath(maybe_path)
-
-
 def is_fs_path(path: Path) -> bool:
```
Comment (normanrz) [inline webknossos/webknossos/utils.py:215]: We could get rid of this as well, but I actually think `is_fs_path` is more expressive than `not isinstance(path, UPath)`.

**Comment by jstriebel** [comment]
@normanrz Is this PR ready for review?

**Comment by normanrz** [comment]
> @normanrz Is this PR ready for review?

Yes!

File: webknossos/webknossos/dataset/_array.py (hunk 1 above)
Comment (jstriebel) [reply webknossos/webknossos/dataset/_array.py:146]: One might argue that this is also a bug in UPath, but always sending two requests for the is_file check also seems wrong. In any case the fix is good atm!

**Review by jstriebel — APPROVED** [review]
Nice, LGTM 👌

File: webknossos/webknossos/dataset/_array.py (hunk 1 above)
Comment (normanrz) [reply webknossos/webknossos/dataset/_array.py:146]: Yeah. I already have a fix for UPath in the pipeline.

**Pull request merged**